DfE.FindInformationAcademiesTrusts/Pages/Trusts/Academies/AcademiesPageModel.cs

```diff
@@ -0,0 +1,38 @@
+﻿using DfE.FindInformationAcademiesTrusts.Data;
+using DfE.FindInformationAcademiesTrusts.Services;
+using DfE.FindInformationAcademiesTrusts.Services.DataSource;
+using DfE.FindInformationAcademiesTrusts.Services.Trust;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DfE.FindInformationAcademiesTrusts.Pages.Trusts.Academies
+{
+    public abstract class AcademiesPageModel(
+        ITrustProvider trustProvider,
+        IDataSourceService dataSourceService,
+        ITrustService trustService,
+        IExportService exportService,
+        ILogger<AcademiesPageModel> logger
+    ) : TrustsAreaModel(trustProvider, dataSourceService, trustService, logger, "Academies in this trust")
+    {
+        protected IExportService ExportService { get; } = exportService;
+
+        public string? TabName { get; init; }
+        public virtual async Task<IActionResult> OnGetExportAsync(string uid)
+        {
+            TrustSummaryServiceModel? trustSummary = await TrustService.GetTrustSummaryAsync(uid);
+            Trust? allAcademiesDetails = await TrustProvider.GetTrustByUidAsync(uid);
+
+            if (trustSummary == null || allAcademiesDetails == null)
+            {
+                return new NotFoundResult();
+            }
+
+            var fileContents = ExportService.ExportAcademiesToSpreadsheetUsingProvider(allAcademiesDetails, trustSummary);
+            string fileName = $"{allAcademiesDetails.Name}-{DateTime.Now.Day}-{DateTime.Now.Month}-{DateTime.Now.Year}.xlsx";
```

Comment:
It would be nice to use a DateTime wrapper for more control in testing. We're unlikely to have problems with this particular code now but this is a good time to introduce the practice of using a wrapper and having it consistently used across the application.

Reply:
I agree that introducing a DateTime wrapper is a good idea! Especially when considering testability and flexibility across the application. I'll implement an IDateTimeProvider that we can inject into this method, and we can start using it consistently across the application. (Not to be confused with old providers, this is the convention for the wrapper haha!)


Whilst I'll implement it here the work to incorporate it will be a technical debt Item I will add for us to review. 

This is done!